experimental/FTheoryTools/src/auxiliary.jl

```diff
@@ -525,3 +525,51 @@ true
 function _martins_desired_blowup(v::NormalToricVarietyType, I::MPolyIdeal; coordinate_name::Union{String, Nothing} = nothing)
   return _martins_desired_blowup(v, ideal_sheaf(v, I))
 end
+
+
+
+###########################################################################
+# 10: Apply a function to the innermost nested structure
+###########################################################################
+
+function deepmap(f, x)
+    if x isa AbstractArray
+        return map(e -> deepmap(f, e), x)
+    else
+        return f(x)
+    end
+end
+
+
+
+###########################################################################
+# 11: Macro for function generation
+###########################################################################
+
+macro define_model_attribute_getter(arg_expr, doc_example="")
+  if !(arg_expr isa Expr && arg_expr.head == :tuple && length(arg_expr.args) == 2)
+    error("Expected input like: (function_name, ReturnType)")
+  end
+
+  fname_expr = arg_expr.args[1]
+  rettype_expr = arg_expr.args[2]
+  fname = fname_expr isa Symbol ? fname_expr : eval(fname_expr)
+  sym = QuoteNode(fname)
+  msg = "No $(replace(string(fname), '_' => ' ')) known for this model"
+
+  default_doc = """
+  Returns `$(fname)` of the F-theory model if known, otherwise throws an error.
+
+  See [Literature Models](@ref "Literature Models") for more details.
+
+  $doc_example
+  """
```

Comment:
This docstring is missing some of the things we usually have. The first line should contain the method signature, indented by 4 spaces. Between the explanation and the docstring, we have `# Examples`

Reply:
Another issue here: In the case that `doc_example` is empty, you still add two empty lines to the docstring. Maybe instead default the second argument of the macro to `nothing` and then depending on `isnothing(doc_example)` use one of two different docstring templates